### DISCUSSION TITLE
How to use multiple externalType?

### DISCUSSION THREAD

**Discussion by liorosh** [webpack/webpack#13441] [Q&A]
I'm trying to use the new `externalType:'script'` feature in Webpack.
Is it possible to use multiple types of externals?
following is an example:
I have a bootstrapping code that runs before my actual applications, it makes sure to fetch the assets and some common resources(vendors) used by all of them, the common resources are placed in a specific namespace on the global window object.
Note that some of my vendors are private packages, they all come as part of my vendor.js bundle.
Each application uses the external object in the `webpack.config` and points to the specific global object as follows:

my bootstrapping code fetches react and places it in:
```javascript
//vendor.js
module.exports={
  ...window.lib,
  react:require('react')
}

```

`app.js` configuration:
```javascript
// webpack.config.js
module.exports={
  ...,
  externals:{
    react:'window.lib.react'
  }
}
```

so far so good, that works as expected.

Now, my actual application uses a private package that in turn uses a 3rd party library, its usage is very low and it makes no sense for me to put it inside the bundle.
Because it's a dependency inside a package dynamic import is not an option. so i'm trying to use the new `externalTypes:'script'` and point that external to a CDN, but that causes issues with the rest of my `var` type externals.

the flow is: react application uses my own package (call it `myPackage`), and `myPackage` uses a 3rd party library called [react-facebook](https://github.com/seeden/react-facebook).
I want to fetch react-facebook from CDN once needed and not bundle it in `myPackage`.

`myPackage.js` configuration:
```javascript
// webpack.config.js
module.exports={
  ...
  externalsType:'script'
  externals:{
    react:'window.lib.react'
    react-facebook:
      'reactFacebook@https://someCdn.com',
  }
}
```
this causes error in the bundles code as shown in the image below, notice lines 607 and 609 with `typeof` of nothing
![image](https://user-images.githubusercontent.com/17067819/119471984-7bfbce80-bd52-11eb-9b1b-0d7befa964f8.png)

Is there a way to use both types of externals?or any sort of combination of externals types?

**Comment by alexander-akait** [comment]
Why do not use:
```
externals: {
    react: 'window lib.react',
    'react-facebook': 'script reactFacebook@https://someCdn.com',
},
```
